crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2653,6 +2654,14 @@ impl KnownFunction {
     }
 }
 
+#[salsa::interned]
+pub struct ModuleLiteralType<'db> {
+    /// The name of the module being referenced
+    pub name: ModuleName,
```

Comment:
I don't have a good sense for what's better but we could remove the `name` here and instead call `file_to_module` and then retrieve the name from `module.name`. We can add a `name(db)` method that hides the `file_to_module` and `unwrap` calls. 

It would reduce the amount of data that we store, possibly at the cost that module attribute lookups are slightly slower.

Reply:
Ah, actually this tells me we should just use `Module` here!  I've kept the `ModuleLiteralType` wrapper struct so that I could mark it as `salsa::tracked`, which lets `Type` still be `Copy`